app/actions/views/channel.js

```diff
@@ -734,6 +739,10 @@ export function loadChannelsForTeam(teamId, skipDispatch = false, isReconnect =
                 }
             }
 
+            if ((lte(config.Version, '5.31.0') && config.ExperimentalChannelSidebarOrganization === 'true') || config.EnableLegacySidebar === 'false') {
+                await dispatch(fetchMyCategories(teamId));
+            }
+
```

Comment:
Only calls `fetchMyCategories` if:
- Server <= 5.31.0 AND ExperimentalChannelSidebarOrganization
- EnableLegacySidebar is false